src/bmffimage.cpp

```diff
@@ -0,0 +1,330 @@
+// ***************************************************************** -*- C++ -*-
+/*
+ * Copyright (C) 2021 Exiv2 authors
+ * This program is part of the Exiv2 distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, 5th Floor, Boston, MA 02110-1301 USA.
+ */
+
+// *****************************************************************************
+
+#define EXIV2_DEBUG_MESSAGES
+
+// included header files
+#include "config.h"
+
+#include "bmffimage.hpp"
+#include "tiffimage.hpp"
+#include "image.hpp"
+#include "image_int.hpp"
+#include "basicio.hpp"
+#include "error.hpp"
+#include "futils.hpp"
+#include "types.hpp"
+#include "safe_op.hpp"
+#include "unused.h"
+
+// + standard includes
+#include <string>
+#include <cstring>
+#include <iostream>
+#include <cassert>
+#include <cstdio>
+
+struct BmffBoxHeader
+{
+    uint32_t length;
+    uint32_t type;
+};
+
+#define ID(string) ((string[0] << 24) | (string[1] << 16) | (string[2] << 8) | string[3])
+
+#define TAG_ftyp ID("ftyp") /**< File type box */
+#define TAG_avif ID("avif") /**< AVIF */
+#define TAG_heic ID("heic") /**< HEIF */
```

Comment:
Typo: HEIF->HEIC

Reply:
These are different tags for files already produced. So no typo, even if it seems obvious.

heic means container
heif means format
heix is extended precision, i.e 10 bits per channel, however I haven't seen such files yet.

Reply:
The Canon HEIV files that Gordon sent have brand: heix.  I'm hoping to setup the test environment tomorrow (2021-02-26).  I don't want to drop big fat files into test/data in the repos.  The test files will be stored somewhere in GitHub and downloaded when cmake -DEXIV2_ENABLE_BMFF=On is used.

Reply:
Bingo! I was waiting for a first encounter with 10 bpc files.